src/services/services.ts

```diff
@@ -2428,6 +2432,50 @@ module ts {
                 return result;
             }
 
+            function isNewIdentifierDefinitionLocation(previousToken: Node): boolean {
+                if (previousToken) {
+                    var containingNodeKind = previousToken.parent.kind;
+                    switch (previousToken.kind) {
+                        case SyntaxKind.CommaToken:
+                            return containingNodeKind === SyntaxKind.CallExpression                         // func( a, |
+                                || containingNodeKind === SyntaxKind.Constructor                            // constructor( a, |
+                                || previousToken.parent.parent.parent.kind === SyntaxKind.VariableDeclaration;     // var x = (a, b| <- this can be a lambda expression
+              
+                        case SyntaxKind.OpenParenToken:
+                            return containingNodeKind === SyntaxKind.CallExpression                         // func( |
+                                || containingNodeKind === SyntaxKind.Constructor                            // constructor( |
+                                || previousToken.parent.parent.kind === SyntaxKind.VariableDeclaration;     // var x = (a| <- this can be a lambda expression

```

Comment:
Instead of this variable declaration case, you may just want to do this for parenthesized expressions